autoload/lsp/ui/vim/signature_help.vim

```diff
@@ -51,6 +83,33 @@ function! s:handle_signature_help(server, data) abort
     endif
 endfunction
 
+function! s:get_parameter_label(signature, parameter) abort
+    if has_key(a:parameter, 'label')
+        if type(a:parameter['label']) == v:t_list
+            let l:string_range = a:parameter['label']
+            return strcharpart(
+                        \ a:signature['label'],
+                        \ l:string_range[0],
+                        \ l:string_range[1] - l:string_range[0])
+        endif
+        return a:parameter['label']
+    endif
+    return ''
+endfunction
+
+function! s:get_parameter_doc(parameter) abort
+    if !has_key(a:parameter, 'documentation')
+        return ''
+    endif
+
+    if type(a:parameter['documentation']) == v:t_dict
+        let l:doc = copy(a:parameter['documentation'])
+        let l:doc['value'] = printf('***%s*** - %s', a:parameter['label'], l:doc['value'])
+        return l:doc
```

Comment:
Actually, I'm wondering what happens if this is passed to `lsp#ui#vim#output#preview`: no case in `s:append` would apply, so these lines will be skipped. Should we not simply return `printf(...)` here?

Reply:
Parameter's documentation is `string or MarkupContent` in LSP spec.

I checked `lsp#ui#vim#output.vim`, it will handle `MarkupContent`.
So I was not modified return format.
(https://github.com/prabirshrestha/vim-lsp/blob/master/autoload/lsp/ui/vim/output.vim#L397)

Though if we return string, `output.vim` still handles documentation as markdown.

Reply:
Ah nevermind, I missed that `MarkupContent` has a `kind`, so the last case in `s:appends` does indeed apply. Sorry for the confusion!

Reply:
No worries.
You code review is very helpful for me! Thanks.